consul/pool.go

```diff
@@ -280,24 +288,51 @@ func (p *ConnPool) DialTimeout(dc string, addr net.Addr, timeout time.Duration)
 
 	// Check if TLS is enabled
 	if p.tlsWrap != nil {
-		// Switch the connection into TLS mode
-		if _, err := conn.Write([]byte{byte(rpcTLS)}); err != nil {
-			conn.Close()
+		doWrap, err := memberExpectsTLS(addr.String(), p.memberFunc())
```

Comment:
This does a lot of magic. Please add a comment explaining what happens here and why.